hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioPipeline.java

```diff
@@ -112,42 +115,29 @@ public NioThread owner() {
     }
 
     void start() {
-        ownerAddTaskAndWakeup(new NioPipelineTask(NioPipeline.this) {
-            @Override
-            protected void run0() {
-                try {
-                    getSelectionKey();
-
-                    // we need to call process so that the pipeline gets unscheduled.
-                    if (NioPipeline.this instanceof NioOutboundPipeline) {
-                        NioOutboundPipeline out = (NioOutboundPipeline) NioPipeline.this;
-                        out.process();
-                    }
-                } catch (Throwable t) {
-                    onError(t);
-                }
+        owner.addTaskAndWakeup(() -> {
+            try {
+                initSelectionKey();
+                process();
+            }catch (Throwable t){
+                onError(t);
             }
         });
     }
 
-    private SelectionKey getSelectionKey() throws IOException {
-        if (selectionKey == null) {
-            selectionKey = socketChannel.register(owner.getSelector(), initialOps, this);
-        }
-        return selectionKey;
+    final void initSelectionKey() throws ClosedChannelException {
```

Comment:
We get rid of the lazy creation of the selectionkey. The selectionkey is is immediately made when the NioPipeline is registered at some NioThread.